### ISSUE TITLE
[Bug]: Songs added from the same album on a different date move next to eachother on Date Added sort

### ISSUE THREAD

**Issue by kobaltkloud** [ciderapp/Cider-2#29]
### Preflight Checklist

- [X] I agree to follow the [Code of Conduct](https://cider.sh/legal/conduct) that this project adheres to.
- [X] I have searched the [issue tracker](https://github.com/ciderapp/Fujisan-BugTracker/issues) for a bug report that matches the one I want to file, without success.
- [X] I have read the [troublshooting FAQs](https://cider.gitbook.io/welcome-to-gitbook/support/faqs) and none solved my issue. (* These are outdated and may not necessarily be possible in the current app version. *)

### What operating system are you using?

Windows (Only Supported)

### Operating System Version

Windows 11 22H2

### Describe the Bug

Songs that are added should always stay in order according to the date they were added, even if they are in the same album. Currently, when a new song is added, the other songs from the same album **added on an earlier date** move up next to the recently added song.
Here is a comparison of the problem between iTunes and Cider.

![image](https://user-images.githubusercontent.com/58888207/223337545-88a7e5a9-61e3-4d74-87f9-3b78916b86f5.png)
_iTunes_
I have added this song from the album "Out Run" and it stays alone even though there are other songs from this album in my library added on an earlier date.

![image](https://user-images.githubusercontent.com/58888207/223338443-23644a2c-002c-4736-aedd-0b17a0eeab4a.png)
_Cider_
Here we can see that the songs added earlier from the album "Out Run" moved up next to the recently added song because they are in the same album.

This behavior messes up the order of the library and when it occurs between a large number of songs and albums, it can get really difficult to find a song the user is looking for without searching the name.

### Steps to Reproduce

1. Sort library in Date Added
2. Add X song
3. Add Y song (from a different album)
4. Add Z song from the same album as the X song is in

The X song takes over the Y song and moves next to the Z song (because they are in the same album)

### Anything else?

- This issue is present in both Cider and Cider 2.
- I would like to acknowledge that this behavior might be better for some users, however, it is critical for me to at least have the option to change it in the settings.

**Comment by coredev-uk** [comment]
So unfortunately it seems we can't really work around this. Apple only provides the raw date added for the album of the library item, not the song itself. The only other alternative option is to make an api call for the list with a sorting option passed. This would not be ideal as it would not be as seamless as sorting is now, and will cause it to take a few seconds to load the sorted page. 

I'll keep looking for an alternative option, but for now it seems this is all we can do.

**Comment by kobaltkloud** [comment]
**Seemed to be fixed on my first 10 minutes on 2.1.1**, checked About to find my version number and when I came back to Songs, Date Added wasn't working at all. The songs are all over the place, no idea what its trying to sort them by. (both on ascending and descending) Date Added itself still broken after restart. Other sorting options seem to be working fine.

TL;DR: This issue was fixed for a short period of time just after installing before it broke completely shortly after.
